internal/core/operations/remove_rooms/remove_rooms_executor.go

```diff
@@ -63,15 +68,21 @@ func (e *RemoveRoomsExecutor) Execute(ctx context.Context, op *operation.Operati
 		if err != nil {
 			reportDeletionFailedTotal(op.SchedulerName, op.ID)
 			logger.Warn("failed to remove rooms", zap.Error(err))
-			return fmt.Errorf("failed to remove room: %w", err)
+			deleteErr := fmt.Errorf("failed to remove room: %w", err)
+
+			if errors.Is(err, serviceerrors.ErrGameRoomStatusWaitingTimeout) {
+				return operations.NewErrTerminatingPingTimeout(deleteErr)
+			} else {
```

Comment:
I think this else is unnecessary, no? Like I imagine we could write more else ifs in case we map more kinds of error, but they would always fallback to "return unexpected"

Reply:
[7d6d6d9](https://github.com/topfreegames/maestro/pull/372/commits/7d6d6d9e568f1e7317e7cba0007b773a5e989935)